### PR TITLE
Refactor decorators

### PR DESCRIPTION
**Pull request by jgroth**
This deprecates all code in the `decorators` package and moves it to the `lime-web-components` package. GitHub does not give a good preview of the changes here, but I followed the same pattern as in #377 

1. One commit that moves the file to the new location
2. A new commit that fixes any broken imports, documentation, and naming issues. Old decorator is deprecated and re-exports the moved code under the old name.

The most notable change in this PR is a new naming convention for the decorators. Previously we had issues with name collision between types from `interfaces` and `decorators`, e.g. we both had a type and a decorator named `Session` so it was difficult to do something like the following without re-naming the types in your own code.

```ts
class MyComponent {
    @State()
    @Session() // The session decorator
    private session: Session; // The type for the session (name collision)
}    
```

With this PR all decorators are now named `Select...`, e.g. `@SelectSession()` and `@SelectCurrentLimeType()`. This was taken from Redux which has the selector concept[^1] of selecting data from the redux store. The `NgRedux` package that we have used in the webclient also has something similar with the generic `@Select` decorator that allows you to select anything from the state so I think this naming convention makes sense.

[^1]: https://redux.js.org/usage/deriving-data-selectors

### CODE REVIEW

File: packages/lime-web-components/src/core/application.ts [hunk 1]
```diff
@@ -10,7 +10,9 @@ import {
  * @param {StateOptions} [options] options for the state selector
  * @returns {Function} state decorator
  */
-export function SelectApplicationName(options: StateOptions = {}) {
+export function SelectApplicationName(
+    options: StateOptions = {}
+): PropertyDecorator {
```
Comment (adrianschmidt) [inline packages/lime-web-components/src/core/application.ts]: Looks like this change belongs in the preceding commit?

File: packages/decorators/src/index.ts [hunk 2]
```diff
@@ -7,4 +7,9 @@ export * from './state/limeobjects';
 export * from './state/limetypes';
```
Comment (adrianschmidt) [inline packages/decorators/src/index.ts:7]: Same thing here as in #377, regarding the commit messages, and making sure the changes appear in the changelog.

**Review by adrianschmidt — CHANGES_REQUESTED** [review]
Two tiny things, otherwise it looks great 👍

File: packages/lime-web-components/src/core/application.ts (hunk 1 above)
Comment (jgroth) [reply packages/lime-web-components/src/core/application.ts]: It took a few tries but I think I managed to sort it out :smile:

File: packages/decorators/src/index.ts (hunk 2 above)
Comment (jgroth) [reply packages/decorators/src/index.ts:7]: Fixed :slightly_smiling_face:

**Review by adrianschmidt — APPROVED** [review]
Looks good to me 👍

**Pull request merged**